src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,155 @@
+    grant_e['operationParam']['operationFrequencyRange']['highFrequency'] = 3570
+    request = {'grantRequest': [grant_c, grant_d, grant_e]}
+    # send grant request and get response
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), 3)       
+    grant_ids = (response[0]['grantId'], response[1]['grantId'], response[2]['grantId'])
+    del request, response
+    # First Heartbeat
+    first_HB_response = self.sendAndAssertSuccessfulHeartBeat(cbsd_ids, grant_ids, [None, None, None])
+    transmitExpireTime = (first_HB_response[0]['transmitExpireTime'], first_HB_response[1]['transmitExpireTime'], first_HB_response[2]['transmitExpireTime'])
+    # STEP 2
+    # load and inject FSS data
+    fss_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'fss_record_c.json')))
+    # Inject Incumbent Activity with Overlapping Frequency of CBSD
+    fss_c['deploymentParam']['operationParam']['operationFrequencyRange']['lowFrequency'] = \
+                             grant_c['operationParam']['operationFrequencyRange']['lowFrequency']
+    fss_c['deploymentParam']['operationParam']['operationFrequencyRange']['highFrequency'] = \
+                             grant_c['operationParam']['operationFrequencyRange']['highFrequency']
+    
+    self._sas_admin.InjectFss({'record': fss_c})   
+    # load and inject GWPZ data
+    gwpz_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'gwpz_record_c.json')))
+    # Inject Incumbent Activity with Overlapping Frequency of CBSD
+    gwpz_c['deploymentParam']['operationParam']['operationFrequencyRange']['lowFrequency'] = \
+                             grant_c['operationParam']['operationFrequencyRange']['lowFrequency']
+    gwpz_c['deploymentParam']['operationParam']['operationFrequencyRange']['highFrequency'] = \
+                             grant_c['operationParam']['operationFrequencyRange']['highFrequency']
+    
+    self._sas_admin.InjectWisp({'record': gwpz_c})      
+    # STEP 3
+    # Trigger daily activities and wait for it to get it complete
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+    # Send Heartbeat after triggering the IAP
+    request =  self.getHeartBeatRequest(cbsd_ids, grant_ids, transmitExpireTime)
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # CHECK
+    for response_num, resp in enumerate(response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        grant_expire_time_2 = datetime.strptime(resp['grantExpireTime'],'%Y-%m-%dT%H:%M:%SZ')
+        transmit_expire_time_2 = datetime.strptime(resp['transmitExpireTime'],'%Y-%m-%dT%H:%M:%SZ')
+        self.assertLessEqual((transmit_expire_time_2 - datetime.utcnow()).total_seconds(), 240)
+        self.assertLessEqual(transmit_expire_time_2, grant_expire_time_2)
+        if cbsd_ids[response_num] == cbsd_ids[2]:
```

Comment:
There's an easier way to write this condition, although normally we would split this block of code into two pieces to make it clearer:

// Checks for CBSDs 1, 2
// Check for CBSD 3 (since it's different)

Reply:
fixed, thank you